### ISSUE TITLE
Drop file upload fails for fullscreen Chrome on OSX (drag across desktops)

### ISSUE THREAD

**Issue by haosdent** [RocketChat/Rocket.Chat#1579]
Could not drop file upload in Chrome while it is OK in Safari.

**Comment by haosdent** [comment]
```
--- a/packages/rocketchat-ui/views/app/room.coffee
+++ b/packages/rocketchat-ui/views/app/room.coffee
@@ -367,6 +367,10 @@ Template.room.events
        'dragleave .dropzone-overlay': (e) ->
                e.currentTarget.parentNode.classList.remove 'over'

+       'dragover .dropzone-overlay': (e) ->
+               e = e.originalEvent or e
+               e.dataTransfer.dropEffect = 'move'
+
        'dropped .dropzone-overlay': (event) ->
                event.currentTarget.parentNode.classList.remove 'over'
```
Need add this to avoid drag upload problem in chrome

**Comment by geekgonecrazy** [comment]
@haosdent looks like you have the solution.  Would be open to submitting a Pull Request for this?

**Comment by Sing-Li** [comment]
But this does not appear to be a problem.  At least not on Windows Chrome, OSX Chrome or Ubuntu Linux Chrome?     @haosdent  - can you explain how to reproduce the 'problem' ?

**Comment by haosdent** [comment]
I use OS X 10.10.5 + Chrome

**Comment by marceloschmidt** [comment]
@haosdent do you experience this problem on the demo server?

**Comment by haosdent** [comment]
yes. The problem exists in demo server.

On Mon, Dec 7, 2015 at 9:11 PM, Marcelo Schmidt <notifications@github.com>
wrote:

> @haosdent <https://github.com/haosdent> do you experience this problem on
> the demo server?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/RocketChat/Rocket.Chat/issues/1579#issuecomment-162520071>
> .
>



-- 
Best Regards,
Haosdent Huang

**Comment by Sing-Li** [comment]
@haosdent   Please click  on Chrome (Right Hamburger menu) --->  Help -> About Google Chrome and get the version:

Mines is  Version 47.0.2526.73 (64-bit)

What's your?  Thanks.

**Comment by haosdent** [comment]
I am
Version 47.0.2526.73 (64-bit)

Google Chrome is up to date.

On Tue, Dec 8, 2015 at 10:38 PM, SingLi <notifications@github.com> wrote:

> @haosdent <https://github.com/haosdent> Please click on Chrome (Right
> Hamburger menu) ---> Help -> About Google Chrome and get the version:
>
> Mines is Version 47.0.2526.73 (64-bit)
>
> What's your? Thanks.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/RocketChat/Rocket.Chat/issues/1579#issuecomment-162901336>
> .
>



-- 
Best Regards,
Haosdent Huang

**Comment by haosdent** [comment]
The record if not apply this patch
![could_not_upload](https://cloud.githubusercontent.com/assets/950759/11660285/6d17ad0a-9e07-11e5-8fca-7a6446418b16.gif)

**Comment by haosdent** [comment]
If apply this patch, it could drop and upload, notice I `tab` when between select file and drop

**Comment by haosdent** [comment]
Another case which drop file directly.

![could_not_upload](https://cloud.githubusercontent.com/assets/950759/11660376/dea60ca0-9e07-11e5-8ca3-490accb02eb6.gif)

**Comment by haosdent** [comment]
And kind in mind, if I not full screen finder and drop, it could handle success.
![could_not_upload](https://cloud.githubusercontent.com/assets/950759/11660420/09eb5eec-9e08-11e5-841f-ccf66af5042d.gif)

**Comment by haosdent** [comment]
If apply mine patch, all these cases could drop and upload. Contains paste img from clipboard

**Comment by Sing-Li** [comment]
Ah!  Thanks @haosdent -  that's POWER USER drag-and-drop on OSX  ;-)

**Comment by haosdent** [comment]
I quite like create a pull request, but I busy on other thing. And this is a small thing. I think you could help commit directly.

**Comment by engelgabriel** [comment]
Closed by PR #1598
